GameController.qml

```diff
@@ -35,6 +36,22 @@ Item {
                 model.move(down - 1, index, 1);
                 break;
         }
+
+             checkWinner()
+        return true;
+    }
+
+    function checkWinner() {
+        for (var i = 0; i < model.count; ++i) {
+            if (model.get(i).value - 1 !== i)
```

Comment:
Hard to understand condition.
Try to check if next element is just bigger than a previous one (so, you should check if array is sorted)